### ISSUE TITLE
Components and patterns that need a guidance change to meet WCAG 2.1

### ISSUE THREAD

**Issue by CharlotteDowns** [alphagov/govuk-frontend#4038]
## What
We need to change our examples or add improved content to our guidance pages in order to resolve issues from the DAC Audit May 2023.

### WCAG criteria
- **Step-by-step navigation**, [Step-by-step navigation 'or' headings must be nested under their parent step headings alphagov/govuk-frontend#2794](https://github.com/alphagov/govuk-design-system/issues/2794) and usability criteria [Step-by-step navigation does not convey all of the information for screen reader users when viewing them as forms alphagov/govuk-design-system#3021](https://github.com/alphagov/govuk-design-system/issues/3021)
- **Password (show/hide)**, [Pages with 2+ password fields must use different programmatic labels for 'show' and 'hide' buttons on those password fields alphagov/govuk-frontend#2795](https://github.com/alphagov/govuk-design-system/issues/2795)
- **Tabs**, [Tab components should show visible headings for all sections on mobile screen sizes alphagov/govuk-frontend#2797](https://github.com/alphagov/govuk-design-system/issues/2797)
- **Links**, [Links to Welsh versions of content should be understandable even when out of context alphagov/govuk-frontend-docs#207](https://github.com/alphagov/govuk-design-system/issues/2796)
### Usability criteria
- **Error message**, **Error summary**, [Error messages should use the same label text as the input when referring to an input alphagov/govuk-frontend#2801](https://github.com/alphagov/govuk-design-system/issues/2801)
- **Pagination**, [Pagination for smaller numbers of pages does not show current page number alphagov/govuk-frontend#3682](https://github.com/alphagov/govuk-frontend/issues/3682)
- **Character count**, [Character count is hidden until you approach a character threshold alphagov/govuk-frontend#3689](https://github.com/alphagov/govuk-frontend/issues/3689)

### Further documentation

[DAC audit 2023 results](https://docs.google.com/spreadsheets/d/1qiWCRwH7QofQgJ9iDLRSsfETLKlZCyrv7z04qjZurhM/edit?pli=1#gid=0) spreadsheet. 

This work is a continuation from the previous DAC audit cards, alphagov/govuk-frontend#3814, alphagov/govuk-frontend#3592 and alphagov/govuk-frontend#3329. 

## Why

This work supports our efforts towards the Epic: [Update styles, components and patterns to WCAG 2.2 AA alphagov/govuk-frontend#3040](https://github.com/alphagov/govuk-frontend/issues/3040).

As we approach [the publishing date for WCAG 2.2](https://www.w3.org/WAI/standards-guidelines/wcag/new-in-22/), we want to ensure our existing govuk-frontend elements are meeting all WCAG 2.1 AA criteria. The only way to pass WCAG 2.2 AA is to also meet all criteria in WCAG 2.1 AA ([with one exception, where a criterion is being removed](https://www.w3.org/WAI/standards-guidelines/wcag/new-in-22/#changes-from-wcag-21-to-wcag-22)).

We're doing other work to assess where updates are needed to align elements with the 9 new WCAG 2.2 criteria, but we likely don't have capacity to run an internal audit across all of the existing 78 WCAG criteria.
```[tasklist]
### Tasks
- [ ] https://github.com/alphagov/govuk-design-system/issues/2794
- [ ] https://github.com/alphagov/govuk-design-system/issues/3021
- [ ] https://github.com/alphagov/govuk-design-system/issues/2795
- [ ] https://github.com/alphagov/govuk-design-system/issues/2797
- [ ] https://github.com/alphagov/govuk-design-system/issues/2796
- [ ] https://github.com/alphagov/govuk-design-system/issues/2801
- [ ] https://github.com/alphagov/govuk-frontend/issues/3682
- [ ] https://github.com/alphagov/govuk-frontend/issues/3689
- [ ] https://github.com/alphagov/govuk-design-system/issues/2799
- [ ] https://github.com/alphagov/govuk-design-system/issues/3097
- [ ] https://github.com/alphagov/govuk-design-system/issues/3105
```

**Comment by CharlotteDowns** [comment]
I _think_ unless anyone can suggest otherwise that this has now been completed 🎉. I'll close the issue and move to Done 🏁
